examples/src/components/MeasureValueFilterDropdownExample.jsx

```diff
@@ -22,35 +24,75 @@ const measures = [franchiseFeesMeasure, franchiseSalesMeasure];
 
 const attributes = [Model.attribute(locationNameDisplayFormIdentifier).localIdentifier("locationName")];
 
-export class MeasureValueFilterDropdownExample extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            filter: null,
-        };
-    }
+const DropdownButton = ({ isActive, measureTitle, onClick }) => {
+    const className = classNames(
+        "gd-mvf-dropdown-button",
+        "s-mvf-dropdown-button",
+        "gd-button",
+        "gd-button-secondary",
+        "button-dropdown",
+        "icon-right",
+        { "icon-navigateup": isActive, "icon-navigatedown": !isActive },
+    );
+
+    return (
+        <button className={className} onClick={onClick}>
+            {measureTitle}
+        </button>
+    );
+};
+
+DropdownButton.propTypes = {
+    isActive: PropTypes.bool.isRequired,
+    measureTitle: PropTypes.string.isRequired,
+    onClick: PropTypes.func.isRequired,
+};
+
+export class MeasureValueFilterDropdownExample extends React.PureComponent {
+    state = {
+        filter: [],
+        displayDropdown: false,
+    };
 
     onApply = filter => {
-        this.setState({ filter });
+        this.setState({ filter: [filter], displayDropdown: false });
+    };
+
+    onCancel = () => {
+        this.toggleButtonRef = null;
+        this.setState({ displayDropdown: false });
+    };
+
+    toggleDropdown = e => {
+        this.toggleButtonRef = !this.state.displayDropdown ? e.currentTarget : null;
+        this.setState(state => ({ ...state, displayDropdown: !state.displayDropdown }));
     };
 
     render() {
-        const { filter } = this.state;
+        const { filter, displayDropdown } = this.state;
         return (
             <div>
-                <MeasureValueFilterDropdown
-                    onApply={this.onApply}
-                    measureTitle={franchiseSalesMeasure.measure.title}
-                    measureIdentifier={franchiseSalesMeasure.measure.localIdentifier}
-                    filter={filter || null}
+                <DropdownButton
+                    onClick={this.toggleDropdown}
+                    isActive={displayDropdown}
+                    measureTitle="Measure"
                 />
+                {displayDropdown ? (
+                    <MeasureValueFilterDropdown
+                        onApply={this.onApply}
+                        onCancel={this.onCancel}
+                        measureIdentifier={franchiseSalesMeasure.measure.localIdentifier}
```

Comment:
Would it be hard to rename this to `measureLocalIdentifier`? Measure identifier could be a metadata measure identifier and some people could be confused. What do you think about it?

Reply:
It seems that we are using this term in this meaning in multiple other places so let us continue with this unfortunate trend.